[DOCS] add migration notebook to docs format

burtenshaw opened the pull request:
This PR transfer the notebook guide by @frascuchon and @nataliaElv over to the documentation format, which makes it easier to follow.

You can find the page here: http://localhost:8000/argilla-python/guides/how_to_guides/migrate_from_legacy_datasets/

n.b. the page is not available via the menu. @davidberenstein1957 we should figure out navigation presentation for this page.

The guide looks like this:

![image](https://github.com/argilla-io/argilla/assets/19620375/e4e00da8-6117-4f45-9663-1a8c8be9e987)

burtenshaw committed:
docs: add migration notebook to docs format

frascuchon commented on argilla/docs/guides/how_to_guides/migrate_from_legacy_datasets.md:
```diff
@@ -0,0 +1,200 @@
+
+- An argilla 1.* server instance running with legacy datasets.
+- An argilla 2.* server instance running. If you don't have one, you can create one by following the [Argilla installation guide](../../getting_started/installation.md).
+- The `argilla` sdk package installed in your environment.
```
Indeed, the argilla v2 SDK is fully compatible with argilla-server >=1.29. So, maybe we can notify users if they want to use the same argilla-server instance in this process. 

So, users can still see both versions of the dataset (legacy and (Feedback)dataset). Once everything is fine, they can still upgrade the server to v2.

frascuchon commented on argilla/docs/guides/how_to_guides/migrate_from_legacy_datasets.md:
```diff
@@ -0,0 +1,200 @@
+workspace = "demo"
+
+settings_v1 = rg_v1.load_dataset_settings(dataset_name, workspace)
+records_v1 = rg_v1.load(dataset_name, workspace, limit=100, query="_exists_:annotated_by")
```
This is just a sample query. The final version of this guide should remove the `limit` and the `query`
```suggestion
records_v1 = rg_v1.load(dataset_name, workspace)
```

frascuchon commented on argilla/docs/guides/how_to_guides/migrate_from_legacy_datasets.md:
````diff
@@ -0,0 +1,200 @@
+            responses=responses,
+        )
+    ```
+
````
I didn't check it but we can use it with a real dataset during the tutorial QA.
```suggestion
=== "For token classification"

def map_to_record_for_span(data: dict, users_by_name: dict, current_user: rg.User) -> rg.Record:
    suggestions = []
    responses = []

    if data.get("prediction"):
        # Prediction for token classification will be a list of tuple (label, start, end)
        spans = [
            {"start": start, "end": end, "label": label}
            for label, start, end in data["prediction"]
        ]
        agent = data.get("prediction_agent")
        suggestions.append(rg.Suggestion(question_name="labels", value=spans, agent=agent))

    if data.get("annotation"):
        # From data[annotation] and data[annotation_agent]
        spans = [
            {"start": start, "end": end, "label": label}
            for label, start, end in data["annotation"]
        ]
        user_id = users_by_name.get(data["annotation_agent"], current_user).id
        responses.append(rg.Response(question_name="label", value=spans, user_id=user_id))

    if data.get("vectors"):
        # From data["vectors"]
        vectors = [rg.Vector(name=name, values=value) for name, value in data["vectors"].items()]

    return rg.Record(
        id=data["id"],
        fields=data["inputs"],
        # The inputs field should be a dictionary with the same keys as the `fields` in the settings
        metadata=data["metadata"],
        # The metadata field should be a dictionary with the same keys as the `metadata` in the settings
        vectors=vectors,
        # The vectors field should be a dictionary with the same keys as the `vectors` in the settings
        suggestions=suggestions,
        responses=responses,
    )

```

burtenshaw committed:
docs: add information for feedback datasets on existing servers

frascuchon commented on argilla/docs/guides/how_to_guides/migrate_from_legacy_datasets.md:
```diff
@@ -0,0 +1,272 @@
+
+dataset_name = "feedback-dataset"
+old_dataset = old_client.datasets(name=dataset_name)
+new_dataset = new_client.datasets.add(dataset)
```
wrong naming or missing def?
```suggestion
new_dataset = new_client.datasets.add(old_dataset)
```

frascuchon committed:
Merge branch 'feat/v2.0.0' into docs/migration-guide-in-docs-format

davidberenstein1957 committed:
Merge branch 'feat/v2.0.0' into docs/migration-guide-in-docs-format

frascuchon committed:
[DOCS] Update and define all record mapping functions (#5009)

Helper functions for record mapping have been reviewed and tested with
several datasets.

nataliaElv commented on argilla/docs/how_to_guides/migrate_from_legacy_datasets.md:
```diff
@@ -1 +1,287 @@
+    
+        if annotation := data.get("annotation"):
+            user_id = users_by_name.get(data["annotation_agent"], current_user).id
+            responses.append(rg.Response(question_name="spans", value=annotation, user_id=user_id))
```
`question_name` is different for `rg.Suggestion` and `rg.Response`, it should be the same, whether that's `labels` or `spans`.

frascuchon replied:
```suggestion
            responses.append(rg.Response(question_name="labels", value=annotation, user_id=user_id))
```

nataliaElv commented on argilla/docs/how_to_guides/migrate_from_legacy_datasets.md:
```diff
@@ -1 +1,287 @@
+
+dataset_name = "feedback-dataset"
+old_dataset = old_client.datasets(name=dataset_name)
+new_dataset = new_client.datasets.add(dataset)
```
I guess this line should be `new_dataset = new_client.datasets.add(old_dataset)`? But it still throws an error even with the change.

burtenshaw committed:
fix: fix errors in code snippets

frascuchon commented on argilla/docs/how_to_guides/migrate_from_legacy_datasets.md:
````diff
@@ -1 +1,287 @@
+As mentioned above, `FeedbackDataset`'s are compatible with Argilla V2 and do not need to be reformatted. However, you may want to migrate your feedback datasets to the new server so that you can deprecate your Argilla 1.* server. Here is a guide on how to migrate your feedback datasets:
+
+```python
+import argilla.v1 as rg_v1
````
```suggestion
import argilla as rg
```

frascuchon commented on argilla/docs/how_to_guides/migrate_from_legacy_datasets.md:
```diff
@@ -1 +1,287 @@
+        with_responses=True, # (1)
+        with_suggestions=True,
+        with_vectors=True,
+        with_metadata=True,
```
This argument does not exists.
```suggestion
```

burtenshaw committed:
Update argilla/docs/how_to_guides/migrate_from_legacy_datasets.md

Co-authored-by: Paco Aranda <francis@argilla.io>

burtenshaw committed:
Update argilla/docs/how_to_guides/migrate_from_legacy_datasets.md

Co-authored-by: Paco Aranda <francis@argilla.io>

frascuchon commented on argilla/docs/how_to_guides/migrate_from_legacy_datasets.md:
````diff
@@ -1 +1,286 @@
+)
+```
+
+1. The `with_responses`, `with_suggestions`, `with_vectors`, and `with_metadata` flags are used to load the records with the responses, suggestions, vectors, and metadata respectively.
````
This scenario has several problems. 

1. We cannot share resources between clients, or even create new resources from other resources. Those actions will modify the internal state of the original resource, so old_dataset settings will change when the new_dataset is created
2. Copy responses from one server to another requires a revision of response users. User IDs won't exist in the target server so we should provide a user mapping in the same way we do for legacy datasets.

Not sure how common would be this scenario. But I propose to skip this for now and think a bit more about how the SDK should work for this. cc @burtenshaw

frascuchon replied:
Also, suggestions are linked to questions by a question ID. When we do:
```python
new_dataset.records.log(
    list(old_dataset.records(with_responses=False, with_suggestions=True, with_vectors=True)),
)
```
an error will be raised since the error contains wrong question info for the provided dataset.

burtenshaw committed:
docs: drop migration of feedback datasets between servers
